packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx

```diff
@@ -200,13 +201,21 @@ const SharedFolderRow = ({ bucket, handleRename, openSharedFolder, handleDeleteS
     click(e)
   }
 
-  const getUserIds = (users: BucketUser[]): string[] => {
+  const getUserLabels = (users: LookupUser[]): string[] => {
     return users.reduce((acc: string[], user): string[] => {
+      if (user.username !== "") {
+        return user.username ? [...acc, user.username] :  acc
+      }
+
+      if (user.public_address !== "") {
+        return user.public_address ? [...acc, centerEllipsis(user.public_address, 4)] :  acc
```

Comment:
I'd show a little more
```suggestion
        return user.public_address ? [...acc, centerEllipsis(user.public_address, 6)] :  acc
```